src/concrete_solve.jl

```diff
@@ -334,6 +334,211 @@ function DiffEqBase._concrete_solve_adjoint(prob, alg,
     out, adjoint_sensitivity_backpass
 end
 
+function DiffEqBase._concrete_solve_adjoint(prob, alg,
+                                            sensealg::AbstractAdjointSensitivityAlgorithm,
+                                            u0::StaticArrays.SVector, p, originator::SciMLBase.ADOriginator,
+                                            args...; save_start = true, save_end = true,
+                                            saveat = eltype(prob.tspan)[],
+                                            save_idxs = nothing,
+                                            kwargs...)
```

Comment:
this seems excessive, why does it have a completely separate dispatch instead of a few targeted `ArrayInterface.ismutable` checks?